### DISCUSSION TITLE
INVALID_TAXPAYER, UNKNOWN_FILE_ID és egyéb általános kérdések

### DISCUSSION THREAD

**Discussion by Almito420** [nav-gov-hu/M2M#41] [Kérdések] [answered]
Sziasztok!

Postmanból tesztelem az API-t, az aktuális teszt környezetet meghívva.

REST-es végpontokon már nem kapok vissza validációs hibákat, viszont UNKNOWN_FILE_ID jelenik meg a createDocument hívásra.

SOAP-os végpontokon minden abevjababan hibátlan XML-re a INVALID_TAXPAYER hibát kapok vissza.

Valamilyen általános probléma lehet (igazából csak a formátumnak kellene különbségnek lennie, az eredményeknek nem), vagy nem jól csinálok valamit? Utóbbi esetben küldöm a request és response adatokat.

Mivel még nem jutottam el sosem a createDocument-ig, így nem tudom mi lesz a végeredmény. A getDocument-et hívva fogok kapni egy KR fájlt? Ha igen, akkor kinek a kulcsával lesz majd az titkosítva? Regisztrációval minden adózó fog rendelkezni saját kulccsal?

Mi az általános eljárás a csatolmányokat tekintve, minden adózó a saját cert-jével hitelesítje a mellékleteket, vagy az API beküldésnél egyértelműen igazolható a képviseleti jogosultság?

Picit hosszútávon gondolkodva lesz majd publikus kitöltő alkalmazás amit lehet egyéb környezetben futtatni (mint az abevjava)?
Illetve a jelenlegi publikus verzió XML (https://nav.gov.hu/api/declarationForms/anyk) is megmarad, vagy az is kiváltásra kerül?
Van valamilyen roadmapeteket az átállásra?

Tudom, hogy a projekt elején vagytok és még sok a kérdőjel, de ezek nagyon fontos információk a tervezéshez.

Nagyon szépen köszönöm a választ és a munkátokat is!

**Comment by TamasPill** [comment]
REST-es végpontokon már nem kapok vissza validációs hibákat, viszont UNKNOWN_FILE_ID jelenik meg a createDocument hívásra.
_A bizonylat xml fájl feltöltése után a visszakapott fileId-val egyből hívható a createDocument. Ilyenkor nem lenne szabad UNKNOWN_FILE_ID-t kapni, mert az azt jelenti, hogy a file nem található a fájltárolón. Ha továbbra sem működik, akkor kérnék request-response-t._

SOAP-os végpontokon minden abevjababan hibátlan XML-re a INVALID_TAXPAYER hibát kapok vissza.
_Ez egy belső validáció, ami a bizonylaton szereplő adózó élő státuszát ellenőrzi. A minta xml-ben megadott adózóval átment volna ezen az ellenőrzésen, de most egyébként kikapcsolásra is került a validáció, hogy egyszerűbb legyen a tesztelés._

Mivel még nem jutottam el sosem a createDocument-ig, így nem tudom mi lesz a végeredmény. A getDocument-et hívva fogok kapni egy KR fájlt? Ha igen, akkor kinek a kulcsával lesz majd az titkosítva? Regisztrációval minden adózó fog rendelkezni saját kulccsal?
_A KR fájl készítést az M2M rendszer végzi, így a kliensnek nem kell titkosítania sem az xml-t, sem a csatolmányokat. A getDocument a beküldés státuszát és a végén az érkeztetési számot adja vissza._

Mi az általános eljárás a csatolmányokat tekintve, minden adózó a saját cert-jével hitelesítje a mellékleteket, vagy az API beküldésnél egyértelműen igazolható a képviseleti jogosultság?
_A képviseleti jogosultságok az EKNYI (Egységes Képviseleti Nyilvántartás) alapján vannak validálva. Itt a beküldőszemély és a bizonylaton szereplő adózó között kell legyen meghatalmazás/képviseleti jog._

Picit hosszútávon gondolkodva lesz majd publikus kitöltő alkalmazás amit lehet egyéb környezetben futtatni (mint az abevjava)?
_Tervezés alatt van a hibaágak kezelésére egy szerkesztő megoldás, de ez még távlati cél._
Illetve a jelenlegi publikus verzió XML (https://nav.gov.hu/api/declarationForms/anyk) is megmarad, vagy az is kiváltásra kerül?
_Ez is tervezés alatt van._
Van valamilyen roadmapeteket az átállásra?
_Későbbiekben várható ilyen kommunikáció._

**Reply by Almito420** [reply]
Köszönöm szépen a gyors válaszokat! Sorban reagálok:

**Request:**

Végpont (POST): https://m2m-dev.nav.gov.hu/rest-api/1.1/NavM2mCommon/filestoreUploadService
header, message_id: 8fe488ca-2275-11ef-9262-0242ac120002
body, file: a példák között lévő 24T24.xml

**Response**:

{
    "resultMessage": null,
    "fileId": "d0b5c501-f5a1-457d-b1d1-63c6f1035131",
    "virusScanResultCode": "PASSED",
    "result_code": "UPLOAD_SUCCESS"
}

**Request:**

Végpont (GET): https://m2m-dev.nav.gov.hu/rest-api/1.1/NavM2mDocument/documentService/Document/6cb0c56d-583c-4761-854a-a54f05727c65
heder, message_id: 4cd1b0e3-00ec-4fa0-8bb8-b4fabf0a91da


**Response**:

{
    "resultMessage": null,
    "documentStatus": null,
    "documentFileId": "d0b5c501-f5a1-457d-b1d1-63c6f1035131",
    "errors": null,
    "arrivalNumber": null,
    "resultCode": "UNKNOWN_FILE_ID"
}

_Ez egy belső validáció, ami a bizonylaton szereplő adózó élő státuszát ellenőrzi. A minta xml-ben megadott adózóval átment volna ezen az ellenőrzésen, de most egyébként kikapcsolásra is került a validáció, hogy egyszerűbb legyen a tesztelés._

Igen, a teszt XML-el próbálkoztam. Azóta már itt is ugyanazt a hibát kapom:

**SOAP response** (https://m2m-dev.nav.gov.hu/soap-api/1.1/document)

```
<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
    <soap:Header>
        <ns2:NAVm2mSoapHeader xmlns:ns2="https://soap.api.nav.gov.hu/definitions/services/1.1/document/local">
            <correlationId>44b11eec-6df6-4167-80ca-5ea4054aad0a</correlationId>
            <messageId>8fe488ca-2275-11ef-9262-0242ac120002</messageId>
        </ns2:NAVm2mSoapHeader>
    </soap:Header>
    <soap:Body>
        <ns2:GetDocumentStatusResponseElem xmlns:ns2="https://soap.api.nav.gov.hu/definitions/services/1.1/document/local">
            <documentFileId>104531c0-c43f-4bd1-a7ff-9a42fadb4b99</documentFileId>
            <resultCode>UNKNOWN_FILE_ID</resultCode>
        </ns2:GetDocumentStatusResponseElem>
    </soap:Body>
</soap:Envelope>
```

_A KR fájl készítést az M2M rendszer végzi, így a kliensnek nem kell titkosítania sem az xml-t, sem a csatolmányokat. A getDocument a beküldés státuszát és a végén az érkeztetési számot adja vissza._

Rendben van, a beküldés az adózó nevében történik a BKSZ-en keresztül vagy ez ettől teljesen független? Esetleges jogvita esetén az adózó hogyan tudja bizonyítani, hogy pontosan milyen KR fájl került elküldésre, ha külső kulccsal kerül titkosításra a KR konténer? Az elküldött fájlt utólag nem lehet megtekinteni\lekérni egy végponton keresztül (vagy legalább a csatolmányok lenyomatait)?

_A képviseleti jogosultságok az EKNYI (Egységes Képviseleti Nyilvántartás) alapján vannak validálva. Itt a beküldőszemély és a bizonylaton szereplő adózó között kell legyen meghatalmazás/képviseleti jog._

Jelenleg az adózók a saját cégkapu tárhelyüket használják és gépi interfészen keresztül küldik be a KR fájlokat (KRID+jelszó). Ha a beküldést megelőzően megkapjuk egy végponton a KR fájlt, akkor technikailag ez is megoldható lenne? A cégkapu indulása óta minden abevjava mellékletet hitelesítenek a felhasználók, ha jól értem, akkor ebben nem fog változás történni.

+1 kérdés: jelenleg az abevjava támogatja a PDF generálást a forrás XML-ekből. Ezt az alkalmazásunkban megjelenítjük a usereknek, hogy lássák pontosan milyen adatokat adtak meg. A kitöltő megoldás fog tudni valami hasonlót?

A többi válaszotokat köszönöm, akkor egyelőre türelmesen várunk amíg szépen minden a helyére kerül.

Hálásan köszönöm a további válaszokat és segítséget! :)

**Reply by Almito420** [reply]
Szeretnék érdeklődni, hogy mikorra várható válasz a fenti kérdésekre?

Nagyon szépen köszönöm előre is! :)

**Reply by Almito420** [reply]
Szeretnék érdeklődni, hogy mikorra várható válasz a fenti kérdésekre?

Nagyon szépen köszönöm előre is! :)

**Reply by TamasPill** [reply]
Elnézést a késedelemért.
Inline a válaszok:

Rendben van, a beküldés az adózó nevében történik a BKSZ-en keresztül vagy ez ettől teljesen független? **Teljesen függetlenül működik.** 

Esetleges jogvita esetén az adózó hogyan tudja bizonyítani, hogy pontosan milyen KR fájl került elküldésre, ha külső kulccsal kerül titkosításra a KR konténer? **Egy esetleges jogvita esetén a beküldő által az üzenetben elhelyezett, az adózó titkos kulcsával készített aláírással támasztható alá, hogy a fájl megegyezik az adózó által beküldöttel. Emiatt érdemes lehet lementeni a beküldött kérések aláírását. A fájl feltöltésnél az aláírás tartalmazza a file hash-t.**

Az elküldött fájlt utólag nem lehet megtekinteni\lekérni egy végponton keresztül (vagy legalább a csatolmányok lenyomatait)? **Az Ügyfélportálon vissza lehet nézni a beküldött bizonylatokat, de a KR fájl ott sem érhető el.**

Jelenleg az adózók a saját cégkapu tárhelyüket használják és gépi interfészen keresztül küldik be a KR fájlokat (KRID+jelszó). Ha a beküldést megelőzően megkapjuk egy végponton a KR fájlt, akkor technikailag ez is megoldható lenne? A cégkapu indulása óta minden abevjava mellékletet hitelesítenek a felhasználók, ha jól értem, akkor ebben nem fog változás történni. **Jelenleg nem tervezünk ilyen fejlesztést, az M2M API-n feltöltött és sikeresen ellenőrzött bizonylatot az M2M csatornán keresztül javasoljuk beadni.**

Jelenleg az abevjava támogatja a PDF generálást a forrás XML-ekből. Ezt az alkalmazásunkban megjelenítjük a usereknek, hogy lássák pontosan milyen adatokat adtak meg. A kitöltő megoldás fog tudni valami hasonlót? **Az ÁNYK (abevjava) nyomtatvány megjelenítéshez kapcsolódó funkciói (mint a pdf generálás is) belátható időn belül elérhetőek maradnak; ha ebben változás várható, a NAV természetesen tájékoztatni fogja az érintetteket a változásról és az esetlegesen a helyébe lépő új megoldásokról is.**

**Reply by Almito420** [reply]
Köszönöm szépen! Roadmap lenne számomra a következő legfontosabb infó. Mikor várható? :)

Nagyon szépen köszönöm előre is!

**Reply by szg-piller** [reply, marked as answer]
Ha jól értem, a roadmap iránti kérdés az ÁNYK (abevjava) program esetleges teljes körű kivezetésének ütemezésére vonatkozik. Erre vonatkozóan egyelőre nem tudunk tájékoztatást adni, jelenleg egy technológiai tervezőmunka van folyamatban. (Ezen a csatornán keresztül csak az M2M-hez közvetlenül kapcsolódó, _technológiai jellegű_ kérdéseket tudunk megválaszolni, egyéb kérdésekkel kapcsolatban kérjük, forduljanak a NAV hivatalos kapcsolattartási csatornáihoz.)

**Reply by Almito420** [reply]
Köszönöm, megyek tovább a NAV-hoz. Legyen szép napotok!

**Comment by TamasPill** [comment]
A probléma csupán annyi, hogy a file feltöltés után a getFileStatus-t lehet meghívni opcionálisan.
Folyamatilag utána a createDocument következik, ami a bizonylat objektumot létrehozza és elindítja a validációját.
Ezután lehet meghívni a getDocument-et, ami ennek az aktuális állapotát adja vissza.

A többi kérdésre később válaszolunk.

**Reply by Almito420** [reply]
Köszönöm szépen, most sikerült beadni a teszten keresztül! A többi választ várom türelemmel.
